modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/execution.scala

```diff
@@ -0,0 +1,78 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+import Scalaz._
+
+/**
+  * This structure holds the current `Execution` under execution. It carries
+  * information about which `Action`s have been completed.
+  *
+  */
+case class Current(execution: Execution[Action \/ Result]) {
+
+  val isEmpty: Boolean = execution.isEmpty
+
+  val actions: List[Action] = {
+
+    def lefts[L, R](xs: List[L \/ R]): List[L] = xs.collect { case -\/(l) => l }
+
+    lefts(execution.toList)
+
```

Comment:
I'm still hesitating about whether to leave the blank line or not at the end of the function, I guess your opinion sorts it out :)